src/presenter/trip-point-board-presenter.js

```diff
@@ -1,60 +1,151 @@
 
 
 export default class TripPointBoardPresenter{
   #tripEventsListContainer = null;
+  #tripHeaderContainer = null;
 
   #listPresernter = new Map();
-  #zeroPointsPresenter = null;
+
+  #zeroPointsView = null;
+  #addNewTripButtonView = null;
 
   #tripPointsModel = null;
   #offersModel = null;
   #destinationsModel = null;
 
-  #points = null;
+  #filterModel = null;
+  #sortModel = null;
+
+  #filterPresenter = null;
+
   #offers = null;
   #destinations = null;
 
+  #currentNewPoint = null;
 
   constructor(
     {
+      tripHeaderContainer,
       tripEventsListContainer,
+
       tripPointsModel,
       offersModel,
-      destinationsModel
+      destinationsModel,
+
+      filterModel,
+      sortModel,
+
+      filterPresenter
     }
   ){
     this.#tripEventsListContainer = tripEventsListContainer;
+    this.#tripHeaderContainer = tripHeaderContainer;
 
     this.#tripPointsModel = tripPointsModel;
     this.#offersModel = offersModel;
     this.#destinationsModel = destinationsModel;
 
-    this.#points = this.#tripPointsModel.points;
+    this.#filterModel = filterModel;
+    this.#sortModel = sortModel;
+
+    this.#filterPresenter = filterPresenter;
+
     this.#offers = this.#offersModel.offers;
     this.#destinations = this.#destinationsModel.destinations;
 
-    this.#zeroPointsPresenter = new TripPointZeroView();
+    this.#tripPointsModel.addObserver(this.#handleModelEvent);
+    this.#sortModel.addObserver(this.#handleModelEvent);
+    this.#filterModel.addObserver(this.#handleModelEvent);
+
+    this.#addNewTripButtonView = new AddNewTripButtonView({newPoinHandler:this.#newPoinHandler});
+    this.#zeroPointsView = new TripPointZeroView({currentFilter: this.#filterModel.filter});
+
+
   }
 
   init() {
-    this.#renderBoard();
+    render(this.#addNewTripButtonView, this.#tripHeaderContainer);
+    this.#renderBoard(this.points);
+  }
+
+  get points() {
+    return this.#filterPresenter.getPoints();
   }
 
-  #handelPointChange = (updatePoint) => {
-    this.#points.set(updatePoint[0],updatePoint[1]);
-    this.#listPresernter.get(updatePoint[0]).replace(updatePoint);
+  #handleViewAction = (update, actionType, updateType) => {
+    switch (actionType) {
+      case UserAction.UPDATE_POINT:
+        this.#tripPointsModel.updatePoint(update, updateType);
+        break;
+      case UserAction.ADD_POINT:
+        this.#tripPointsModel.addPoint(update, updateType);
+        this.#resetTripPointNew();
+        break;
+      case UserAction.DELETE_POINT:
+        this.#tripPointsModel.deletePoint(update, updateType);
+        break;
+    }
+  };
+
+  #handleModelEvent = (data, updateType) => {
+    switch (updateType) {
+      case UpdateType.PATCH:
+        this.#listPresernter.get(data[0]).replace(data);
+        break;
+      case UpdateType.MINOR:
+        this.#renderBoard(this.points);
+        break;
+      case UpdateType.MAJOR:
+        this.#renderBoard(this.points);
+        break;
```

Comment:
Несколько одинаковых кейсов можно объединять:
```suggestion
      case UpdateType.MINOR:
      case UpdateType.MAJOR:
        this.#renderBoard(this.points);
        break;
```